### PR TITLE
Test to run gsl self test - v2

### PR DESCRIPTION
**Pull request by Pavithra1602**
Script downloads gsl tar ball from ftp://ftp.gnu.org/gnu/gsl/ and runs self test.

Signed-off-by: pavrampu@linux.vnet.ibm.com pavrampu@in.ibm.com

### CODE REVIEW

File: toolchain/gsl.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
+    def test(self):
+        process.run("make -k check", ignore_status=True, sudo=True)
+        logfile = os.path.join(self.logdir, "stdout")
+        failed_tests = process.system_output(
+            "grep -Eai 'FAIL:  [1-9]' %s" % logfile, shell=True, ignore_status=True)
+        if failed_tests:
```
Comment (narasimhan-v) [inline toolchain/gsl.py]: `if process.system_output("grep -Eai 'FAIL:  [1-9]' %s" % logfile, shell=True, ignore_status=True):` is better

File: toolchain/gsl.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        sm = SoftwareManager()
+        dist = distro.detect()
+        packages = ['gcc', 'make']
+        for package in packages:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error("Fail to install %s required for this test." %
+                           package)
+        gsl_version = self.params.get('gsl_version', default='2.2')
+        tarball = self.fetch_asset(
+            "ftp://ftp.gnu.org/gnu/gsl/gsl-%s.tar.gz" % gsl_version)
```
Comment (narasimhan-v) [inline toolchain/gsl.py]: Can you please add a `expire` parameter to `fetch_asset()`

**Comment by PraveenPenguin** [comment]
@Pavithra1602 why  Signed-off-by  with two different ?

File: toolchain/gsl.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(
+            self.srcdir, os.path.basename(tarball.split('.tar')[0]))
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
+    def test(self):
+        process.run("make -k check", ignore_status=True, sudo=True)
```
Comment (PraveenPenguin) [inline toolchain/gsl.py]: can you please use
`retun_log = build.run_make(self.srcdir,extra_args='-k check', ignore_status=True)`

File: toolchain/gsl.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        logfile = os.path.join(self.logdir, "stdout")
```
Comment (PraveenPenguin) [inline toolchain/gsl.py]: when you use retun_log you  need to skip this and log will capture in retun_log.stdout

**Comment by Pavithra1602** [comment]
Raised new PR https://github.com/avocado-framework-tests/avocado-misc-tests/pull/162

**Pull request closed without merging**